src/AutoMapper/TypeExtensions.cs

```diff
@@ -126,7 +127,32 @@ public static bool HasAnInaccessibleSetter(this PropertyInfo property)
 
         public static bool IsValueType(this Type type) => type.GetTypeInfo().IsValueType;
 
-        public static bool IsLiteralType(this Type type) => type == typeof(string) || type.GetTypeInfo().IsValueType;
+        public static bool IsLiteralType(this Type type)
+        {
+            if (PrimitiveHelper.IsNullableType(type))
+                type = Nullable.GetUnderlyingType(type);
+
+            return new HashSet<Type>
```

Comment:
This should be cached and shared somehow with [this](https://github.com/AutoMapper/AutoMapper/blob/c4a091badf28b5851f340998e00f9aa84ad3d322/src/AutoMapper/Mappers/ConvertMapper.cs#L19).

Reply:
Cached the HashSet but those two lists are not the same set of types.

Reply:
Well, yes, but yours is a superset, so cache both and that method checks one and yours both.